### PR TITLE
Massive overhaul and cleanup of how daily smartblocks work (Closes #64)

### LINKED ISSUE #64: Improve stability of daily workflows
- if there are multiple devices with Roam open, it fires twice.
- in some cases, it fails to fire.
- Great solution from [DanB](https://roamresearch.slack.com/archives/C016N2B66JU/p1680920065043529) - add an option to "only run on this device" by generating a device id to store in local storage

### PR DESCRIPTION
**Pull request by dvargas92495**
Next Steps:
- [x] pull out `getDailyConfig` into its own method, use everywhere
- [x] pipe `scheduleNextRun` into roam depot config and sbBomb contexts
- [ ] ~Test multiple devices~ - can only test this with the live version, since there won't be shared settings state.
- [x] Support "only on this device"

### CODE REVIEW

File: lambdas/smartblocks-daily_put.ts [hunk 1]
```diff
@@ -1,44 +0,0 @@
-import { APIGatewayProxyHandler } from "aws-lambda";
```
Comment (dvargas92495) [inline lambdas/smartblocks-daily_put.ts:1]: We used to store whether or not Daily smartblocks ran in RoamJS pre-Roam Depot. No longer necessary

File: lambdas/smartblocks-store_get.ts [hunk 2]
```diff
@@ -31,10 +24,11 @@ const getWorkflow = ({
           TableName: "RoamJSSmartBlocks",
           Item: {
             uuid: { S: nanoid() },
-            name: { S: item.uuid.S },
+            name: { S: item?.uuid.S },
             author: { S: graph },
-            workflow: { S: item.workflow.S },
+            workflow: { S: item?.workflow.S },
             status: { S: toStatus("INSTALLED") },
+            installed: { S: new Date().toJSON() }, // let's see if anyone is actually using this
```
Comment (dvargas92495) [inline lambdas/smartblocks-store_get.ts:31]: this is the only notable change, the rest is typescript cleanup.

I'd like to replace the smartblocks store with SamePage, so I'm adding this field here to track how much this is even getting used until then

File: lambdas/smartblocks-store_put.ts [hunk 3]
```diff
@@ -80,7 +80,7 @@ export const handler: APIGatewayProxyHandler = awsGetRoamJSUser(
       })
```
Comment (dvargas92495) [inline lambdas/smartblocks-store_put.ts:80]: typescript cleanup, ignore

File: src/SmartblocksMenu.tsx [hunk 4]
```diff
@@ -104,21 +111,11 @@ const SmartblocksMenu = ({
                 const title = getPageTitleByBlockUid(blockUid);
                 if (DAILY_NOTE_PAGE_REGEX.test(title)) {
                   const newDate = getPageUidByPageTitle(title);
```
Comment (dvargas92495) [inline src/SmartblocksMenu.tsx:113]: This method contains the only notable change, rest is typescript cleanup

**Review by mdroidian — APPROVED** [review]
Looks good 👍

**Pull request merged**